### ISSUE TITLE
frontend(pgwire): support implicit flush for DMLs with `RETURNING` clause

### ISSUE THREAD

**Issue by BugenZhao** [risingwavelabs/risingwave#7402]
We call `flush_for_write` in the handler of DMLs if the `IMPLICIT_FLUSH` variable is set.

https://github.com/risingwavelabs/risingwave/blob/d956d9c6c33946792e859cd789678095f96abf59/src/frontend/src/handler/query.rs#L200-L205

This is based on the assumption of DMLs are already done, which is true for those without the `RETURNING` clause as the number of affected rows is obtained and it's only yielded after the DML executor has finished. However, DMLs with the `RETURNING` clause are treated as queries, which are polled in the pgwire's logic, `flush` here won't work for it.

A possible fix is to support "post-work" as a boxed future in the `PgResponseStream`, which is called only after all items have been yielded. We can put the calling of `flush_for_write` here.

cc @BowenXiao1999

**Comment by BowenXiao1999** [comment]
Good idea and LGTM. 

BTW, do we need to judge whether it is dml in PostWork?

**Comment by BugenZhao** [comment]
We can either decide in the handler and optionally set this field, or encapsulate this `if `logic in the `async` block.
